### DISCUSSION TITLE
Change config to accept connection with username + password

### DISCUSSION THREAD

**Discussion by mparkhomenko** [strongswan/strongswan#1104] [Q&A]
I have read and configured strongswan with this article https://www.digitalocean.com/community/tutorials/how-to-set-up-an-ikev2-vpn-server-with-strongswan-on-ubuntu-18-04-2

On the client, This one works with username + password + certificate.

I need to change the configuration so the server can accept client connection with just username + password

I a little changed the conf files from a tutorial for using a VPN with a PSK key. At this moment I have a configuration that works with the PSK key. ipsec.secrets:
`someip : PSK "somekey"`

And my ipsec.conf:
```
config setup
        charondebug="ike 1, knl 1, cfg 0"
        uniqueids=no

conn ikev2-vpn
        auto=add
        compress=no
        type=tunnel
        keyexchange=ikev2
        fragmentation=yes
        forceencaps=yes
        ike=aes256-sha1-modp1024,3des-sha1-modp1024!
        esp=aes256-sha1,3des-sha1!
        dpdaction=clear
        dpddelay=300s
        rekey=no
        left=%any
        leftid=ip
        leftsubnet=0.0.0.0/0
        right=%any
        rightid=%any
        rightdns=8.8.8.8,8.8.4.4
        rightsourceip=10.10.10.0/24
        authby=secret
```

What do I need to change in my conf files for use in connection only username + password (username : EAP "password")?

**Comment by tobiasbrunner** [comment]
> What do I need to change in my conf files for use in connection only username + password (username : EAP "password")?

What exactly do you mean? When using EAP client authentication, you still have to authenticate the server in some way (use `left|rightauth` instead of `authby` to configure asymmetric authentication, as the tutorial does). And technically, combining EAP with a PSK is not valid, so you're back to a certificate. However, if you use one from [Let's Encrypt](https://letsencrypt.org/) instead of a self-signed one, most clients will already have the required CA certificate installed.

**Reply by mparkhomenko** [reply]
Thank you for your response! I mean that I want to use a combination with only username + password (without certificate). Because in a tutorial on the digital ocean I need to save a certificate on a client-side, but I don't want to do it. I mean that I want to use a `username : EAP "password"` in my ipsec.secrets without a `: RSA "server-key.pem"`. If it possible

p.s. I created a configuration with PSK secret key and this solution doesn't require a certificate from the client. Can I do the same with username and password?

**Reply by tobiasbrunner** [reply]
Please read my previous answer again.

**Reply by mparkhomenko** [reply]
I understood. Thank you!
